internal/cli/cli.go

```diff
@@ -161,3 +164,46 @@ func configureHealthChecks(mgr manager.Manager) error {
 	}
 	return nil
 }
+
+func getHandlers(resources string, ptm pkgadmission.PodTemplateSpecMutator) ([]admission.AdmissionHandler, error) {
+	var handlers []admission.AdmissionHandler
+	var unexpected []string
+
+	if len(resources) == 0 {
+		return handlers, nil
+	}
+
+	dedupedResources := make(map[string]bool)
+	for _, resource := range strings.Split(resources, ",") {
+		dedupedResources[strings.TrimSpace(resource)] = true
+	}
+
+	for resource := range dedupedResources {
+		switch resource {
+		case "cronjobs":
+			handlers = append(handlers, pkghandlers.NewCronjobHandler(ptm))
+		case "daemonsets":
+			handlers = append(handlers, pkghandlers.NewDaemonSetHandler(ptm))
+		case "deployments":
+			handlers = append(handlers, pkghandlers.NewDeploymentHandler(ptm))
+		case "jobs":
+			handlers = append(handlers, pkghandlers.NewJobHandler(ptm))
+		case "pods":
+			handlers = append(handlers, pkghandlers.NewPodHandler(ptm))
+		case "replicasets":
+			handlers = append(handlers, pkghandlers.NewReplicaSetHandler(ptm))
+		case "replicationcontrollers":
+			handlers = append(handlers, pkghandlers.NewReplicationControllerHandler(ptm))
+		case "statefulsets":
+			handlers = append(handlers, pkghandlers.NewStatefulSetHandler(ptm))
```

Comment:
For handling versioned resources, I'm thinking we can use `<resource>_<version>` suffix.
Example: `statefulsets_v2`

`statefulsets` without suffix is equivalent to `apiVersions: "*"`.

Reply:
A json type dot notation feels more natural.  e.g. v2.statefulset, statefulset (default).

Reply:
I don't think we need to worry about resource version here. We just load all the handlers for a resource kind if the kind is enabled.  The handlers determine if a version is supported. If we don't want a version passed through the controller, we can filter that whole version out in the mutating webhook configuration.

Reply:
That's a great point and reduces complexity both here and in the kanopy module.